src/kudu/security/tls_context.cc

```diff
@@ -300,6 +307,50 @@ Status TlsContext::AddTrustedCertificate(const Cert& cert) {
   return Status::OK();
 }
 
+Status TlsContext::AddTrustedCertificateAltUnlocked(const Cert& cert) {
```

Comment:
This function looks to me like it's only different from the above one in 

```
  X509_STORE *cert_store = X509_STORE_new();
```
and
```
SSL_CTX_set_cert_store(ctx_.get(), cert_store);
```

Maybe we can just do a optional parameter?